src/main/java/ch/njol/skript/expressions/ExprLore.java

```diff
@@ -1,236 +1,181 @@
 package ch.njol.skript.expressions;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-
-import org.bukkit.Bukkit;
-import org.bukkit.Material;
-import org.bukkit.event.Event;
-import org.bukkit.inventory.ItemStack;
-import org.bukkit.inventory.meta.ItemMeta;
-import org.jetbrains.annotations.Nullable;
-
 import ch.njol.skript.Skript;
-import ch.njol.skript.SkriptConfig;
 import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.bukkitutil.ItemUtils;
 import ch.njol.skript.classes.Changer.ChangeMode;
 import ch.njol.skript.classes.Changer.ChangerUtils;
-import ch.njol.skript.doc.Description;
-import ch.njol.skript.doc.Examples;
-import ch.njol.skript.doc.Name;
-import ch.njol.skript.doc.Since;
+import ch.njol.skript.doc.*;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
 import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.registrations.EventValues;
 import ch.njol.util.Kleenean;
-import ch.njol.util.Math2;
-import ch.njol.util.StringUtils;
 import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * TODO make a 'line %number% of %text%' expression and figure out how to deal with signs (4 lines, delete = empty, etc...)
- *
- * @author joeuguce99
  */
 @Name("Lore")
 @Description("An item's lore.")
-@Examples("set the 1st line of the item's lore to \"&lt;orange&gt;Excalibur 2.0\"")
+@Example("""
+	set {_item} to diamond named "Bling Diamond" with lore "&bThe blingiest of diamonds"
+	add "With extra steps" to lore of {_item}
+	remove line 1 of lore of {_item} from lore of {_item}
+	set line 3 of lore of {_item} to "-----"
+	""")
 @Since("2.1")
 public class ExprLore extends SimpleExpression<String> {
 
+	private static final boolean IS_RUNNING_1_21_5 = Skript.isRunningMinecraft(1,21,5);
+
 	static {
-			Skript.registerExpression(ExprLore.class, String.class, ExpressionType.PROPERTY,
-					"[the] lore of %itemstack/itemtype%", "%itemstack/itemtype%'[s] lore",
-					"[the] line %number% of [the] lore of %itemstack/itemtype%",
-					"[the] line %number% of %itemstack/itemtype%'[s] lore",
-					"[the] %number%(st|nd|rd|th) line of [the] lore of %itemstack/itemtype%",
-					"[the] %number%(st|nd|rd|th) line of %itemstack/itemtype%'[s] lore");
+		Skript.registerExpression(ExprLore.class, String.class, ExpressionType.PROPERTY,
+				"[the] lore of %itemstack/itemtype%",
+				"%itemstack/itemtype%'[s] lore",
+				"[the] line %number% of [the] lore of %itemstack/itemtype%",
+				"[the] line %number% of %itemstack/itemtype%'[s] lore",
```

Comment:
```suggestion
				"line %number% of [the] lore of %itemstack/itemtype%",
				"line %number% of %itemstack/itemtype%'[s] lore",
```
We may consider other breaking changes while we are here (`the line 1 of lore of player's tool` does not make sense, right?)